rskj-core/src/test/java/co/rsk/peg/BridgeSupportSvpTest.java

```diff
@@ -969,68 +974,91 @@ void registerBtcTransaction_whenValidationPeriodEnded_shouldNotProcessSpendTx()
             assertActiveFederationUtxosSize(activeFederationUtxosSizeBeforeRegisteringTx + 1);
-            assertTransactionWasProcessed(svpSpendTransaction.getHash());
-            assertNoSvpSpendTxHash();
+            assertTransactionWasProcessed(pegout.getHash());
+            // but spend tx was not
+            assertTransactionWasNotProcessed(svpSpendTransaction.getHash());
+
+            // svp success was not processed
+            assertSvpSpendTxHashUnsignedIsSavedInStorage();
+            assertNoHandoverToNewFederation();
+            assertProposedFederation();
         }
 
         @Test
-        void registerBtcTransaction_whenIsNotTheSpendTransaction_shouldNotProcessSpendTx() throws BlockStoreException, BridgeIllegalArgumentException, IOException {
+        void registerBtcTransaction_whenSpendTransactionHashIsNotSaved_shouldNotProcessSpendTx() throws BlockStoreException, BridgeIllegalArgumentException, IOException {
             // arrange
-            arrangeSvpSpendTransaction();
+            recreateSvpSpendTransactionUnsigned();
             setUpForTransactionRegistration(svpSpendTransaction);
 
-            BtcTransaction pegout = createPegout(proposedFederation.getRedeemScript());
-            savePegoutIndex(pegout);
-            signInputs(pegout);
-            setUpForTransactionRegistration(pegout);
+            int activeFederationUtxosSizeBeforeRegisteringTx = federationSupport.getActiveFederationBtcUTXOs().size();
 
             // act
-            int activeFederationUtxosSizeBeforeRegisteringTx = federationSupport.getActiveFederationBtcUTXOs().size();
             bridgeSupport.registerBtcTransaction(
                 rskTx,
-                pegout.bitcoinSerialize(),
+                svpSpendTransaction.bitcoinSerialize(),
                 btcBlockWithPmtHeight,
                 pmtWithTransactions.bitcoinSerialize()
             );
             bridgeStorageProvider.save();
 
             // assert
-            assertActiveFederationUtxosSize(activeFederationUtxosSizeBeforeRegisteringTx + 1);
+            // spend tx was not registered
+            assertActiveFederationUtxosSize(activeFederationUtxosSizeBeforeRegisteringTx);
             assertTransactionWasNotProcessed(svpSpendTransaction.getHash());
-            assertSvpSpendTxHashUnsignedIsSavedInStorage();
+
+            // svp success was not processed
+            assertNoHandoverToNewFederation();
+            assertProposedFederation();
+        }
+
+        private void assertProposedFederation() {
+            assertTrue(federationSupport.getProposedFederation().isPresent());
         }
 
         @Test
-        void registerBtcTransaction_whenSpendTransactionHashIsNotSaved_shouldNotProcessSpendTx() throws BlockStoreException, BridgeIllegalArgumentException, IOException {
+        void registerBtcTransaction_whenIsTheSpendTransaction_shouldProcessSpendTxAndSvpSuccess() throws BlockStoreException, BridgeIllegalArgumentException, IOException {
```

Comment:
I think we are missing some border cases:

- Trying to register the spend transaction for a second time. What is the expected result?
- Trying to register the spend transaction spent by the ERP Flow. I guess it should be registered if the transaction arrives on time.

Reply:
1) Trying to register the spend transaction for a second time. What is the expected result?
the tx is being marked as processed when registered, same as any other tx. So it won't be registered again

Reply:
Not sure I followed the second part, you need a year to sign ith the ERP flow

Reply:
Truel, the value set is 1 year. But shouldn't we also check that we can spend using the ERP Flow? Just an idea that came to my mind.

Reply:
https://github.com/rsksmart/rskj/pull/2854#discussion_r1854297091